### PR TITLE
Bug 1958097: tokenreviews: improve errors on old-format access tokens

### PR DESCRIPTION
**Pull request by stlaz**
Neither the old format nor serviceaccount tokens have the sha256~
prefix, return a NotFound error in case the tokenvalidation endpoint
is reached with a non-prefixed token.

/assign @sttts 
/cc @s-urbaniak

### CODE REVIEW

**Comment by stlaz** [comment]
/bugzilla refresh
.retest

File: pkg/tokenvalidation/bootstrapauthenticator.go [hunk 1]
```diff
@@ -38,7 +38,12 @@ func NewBootstrapAuthenticator(tokens oauthclient.OAuthAccessTokenInterface, get
 
 func (a *bootstrapAuthenticator) AuthenticateToken(ctx context.Context, name string) (*kauthenticator.Response, bool, error) {
 	if !strings.HasPrefix(name, sha256Prefix) {
-		return nil, false, errOldFormat
+		_, err := a.tokens.Get(ctx, name, metav1.GetOptions{})
```
Comment (sttts) [inline pkg/tokenvalidation/bootstrapauthenticator.go:41]: a comment please for thiis block.

File: pkg/tokenvalidation/bootstrapauthenticator.go [hunk 2]
```diff
@@ -38,7 +38,12 @@ func NewBootstrapAuthenticator(tokens oauthclient.OAuthAccessTokenInterface, get
...
+		if err == nil {
+			return nil, false, errOldFormat
+		} else {
```
Comment (sttts) [inline pkg/tokenvalidation/bootstrapauthenticator.go:44]: remove else parenthesis

Comment (stlaz) [reply pkg/tokenvalidation/bootstrapauthenticator.go]: :+1: how very silly of me

File: pkg/tokenvalidation/bootstrapauthenticator.go [hunk 3]
```diff
@@ -38,7 +38,12 @@ func NewBootstrapAuthenticator(tokens oauthclient.OAuthAccessTokenInterface, get
 
 func (a *bootstrapAuthenticator) AuthenticateToken(ctx context.Context, name string) (*kauthenticator.Response, bool, error) {
 	if !strings.HasPrefix(name, sha256Prefix) {
-		return nil, false, errOldFormat
+		// check that the token exists and report it's of an insecure format if it does
```
Comment (sttts) [inline pkg/tokenvalidation/bootstrapauthenticator.go:41]: would prefer something like:
```
// only complain about non-sha256 format if the token is really an existing OAuthAccessToken (and e.g. no service account which also has no sha256 prefix)
```

**Pull request merged**